### PR TITLE
engine/board.py tweaks

### PR DESCRIPTION
**Pull request by cvennel**
- Added tests to game board
- fixed unresolved reference error in board.py where players should have been self.player
- changed add_player to not be a protected method in Player_queue.py

### CODE REVIEW

File: backend/game/engine/player_queue.py [hunk 1]
```diff
@@ -10,7 +10,7 @@ def __init__(self):
         self.current_player = None
         self.player_cycle = []
 
-    def _add_player_to_game(self, new_player):
```
Comment (cvennel) [inline backend/game/engine/player_queue.py]: @gpfaia I'm not seeing why this was a protected method. If this should be protected let me know and we can change it back

Comment (gpfaia) [reply backend/game/engine/player_queue.py]: No originally just wanted people to be mindful of adding players after the game was started was the original intention

File: backend/game/engine/test_board.py [hunk 2]
```diff
@@ -12,3 +13,53 @@ def test_after_game_starts():
...
+def test_get_next_player_turn_consistent_order():
+    game = BoardGame()
+    for _ in range(3):
+        game.add_player(Player())
+    game.start_game()
+    player1 = game.get_next_player_turn()
+    player2 = game.get_next_player_turn()
+    player3 = game.get_next_player_turn()
+
+    for _ in range(50):
```
Comment (gpfaia) [inline backend/game/engine/test_board.py]: Probably a bit excessive but gets the point across

Comment (cvennel) [reply backend/game/engine/test_board.py]: Yeah, was definitely excessive

**Comment by cvennel** [comment]
@gpfaia  just had to resolve merge conflicts which invalidated your approval. While I was at it, changed that excessive 50 times loop to 5

**Pull request merged**